src/passes/expressionSplitter.ts

```diff
@@ -27,9 +34,35 @@ export class ExpressionSplitter extends ASTMapper {
   visitAssignment(node: Assignment, ast: AST): void {
     this.commonVisit(node, ast);
     if (!(node.parent instanceof ExpressionStatement)) {
-      const replacement = cloneASTNode(node.vLeftHandSide, ast);
-      ast.replaceNode(node, replacement);
-      ast.insertStatementBefore(replacement, new ExpressionStatement(ast.reserveId(), '', node));
+      // No need to create temp vars for state vars
+      if (
+        node.vLeftHandSide instanceof Identifier &&
+        identifierReferenceStateVar(node.vLeftHandSide)
+      ) {
+        return;
+      }
+
+      const leftHandSide = cloneASTNode(node.vLeftHandSide, ast);
+      const rightHandSide = cloneASTNode(node.vRightHandSide, ast);
+
+      const tempVarStatement = createVariableDeclaration(
```

Comment:
Feel that this name should be createVariableDeclarationStatement since it returns a statement and not just a VariableDeclaration.